cmd/jujud/agent/machine/manifolds.go

```diff
@@ -186,6 +188,16 @@ func Manifolds(config ManifoldsConfig) dependency.Manifolds {
 		externalUpdateProxyFunc = lxd.ConfigureLXDProxies
 	}
 
+	newExternalControllerWatcherClient := func(apiInfo *api.Info) (
```

Comment:
It would be nice to be consistent with how the remoterelations worker is set up. That worker's manifold has a func to get the controller connection and then there's a shim.go in the worker package that constructs the client - so we pass in the connection rather than the facade client. I guess the use cases might be subtly different enough that maybe it doesn't make sense to consolidate on a single approach.

Reply:
The idea was to avoid a dependency on the specific API facade, and have that passed in by the code configuring/constructing the manifold. My approach means that this package can just require an interface, and not be concerned at all with how it's fulfilled.